### PR TITLE
capture uncaught exceptions

### LINKED ISSUE #17: Problems with Exceptions when all Windows are closed
I am currently working on #15.
When the Plugin throws an error, between the user pressed connect and the display of the main-window, the Plugin throws that error but does not terminate its process.

This may also occurs in other situations.

### PR DESCRIPTION
**Pull request by manuel3108**
close #17

### CODE REVIEW

File: src/main/java/org/gleisbelegung/ui/lib/window/WindowController.java [hunk 1]
```diff
@@ -20,6 +21,12 @@ public WindowController(WindowInterface window, Stage stage) {
         this.window = window;
 
         panelController = new PanelController();
+
+        Thread.currentThread().setUncaughtExceptionHandler((thread, throwable) -> {
+            Logger.getInstance().uncaughtException(thread, throwable);
+            close();
+            System.exit(1);
```
Comment (Greonyral) [inline src/main/java/org/gleisbelegung/ui/lib/window/WindowController.java]: `exit()` makes testing impossible ->remove it?
Uncaught exceptions will kill the running thread, you must just ensure that the remaining threads notice that they shall also exit, that is normally the idea of `Thread.interrupt()`.

https://docs.oracle.com/javase/8/docs/api/java/lang/Thread.html#enumerate-java.lang.Thread:A- or
https://docs.oracle.com/javase/8/docs/api/java/lang/ThreadGroup.html#enumerate-java.lang.Thread:A-
 and 
https://docs.oracle.com/javase/8/docs/api/java/lang/Thread.html#interrupt-- the Threads you get.

The size for the `Thread[]` can be determined by https://docs.oracle.com/javase/8/docs/api/java/lang/Thread.html#activeCount--

Comment (manuel3108) [reply src/main/java/org/gleisbelegung/ui/lib/window/WindowController.java]: I tried this:
```
Thread.currentThread().setUncaughtExceptionHandler((thread, throwable) -> {
    Logger.getInstance().uncaughtException(thread, throwable);
    close();

    Thread[] threads = new Thread[Thread.activeCount()];
    Thread.enumerate(threads);
    for (Thread t : threads) {
        t.interrupt();
    }
});
```

The problem with this is that there is at least on Thread which is not ending. When I run this in IntelliJ the button to close everything is still available. This was not the case with `System.exit()`, that's why I tried it this way

Comment (Greonyral) [reply src/main/java/org/gleisbelegung/ui/lib/window/WindowController.java]: Ok, if you think that the `exit()` is the better alternative, I won't stop you.

Comment (manuel3108) [reply src/main/java/org/gleisbelegung/ui/lib/window/WindowController.java]: I don't think that `exit()` is the better alternative. But a the moment I did not found another way

**Pull request merged**